### PR TITLE
Website mockup generator

### PR DESCRIPTION
**Pull request by hackyguru**
Adding new workshop - 'Website Mockup Generator'.

Link to the recorded meeting - https://hackclub.slack.com/archives/C01A9V5LPD0/p1609314680060100

### CODE REVIEW

File: workshops/website_mockup_generator/README.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+However, this workshop is beginner friendly and you can refer the [final code][final_code] to understand better.
+
+
+## Part 2: Getting started
+
+### Setting up Repl.it
+
+[Repl.it](https://repl.it) is an amazing online code editor where we will be writing our code. Though repl.it is not mandatory, I strongly recommend you to use it in order to avoid installations and downloads.
+
+To get started, Create your Repl by going to [https://repl.it/languages/html](https://repl.it/languages/html). Your coding environment will be created in a few moments.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:39]:
```suggestion
To get started, create your repl by going to [https://repl.it/languages/html](https://repl.it/languages/html). Your coding environment will be created in a few moments.

```

Comment (faisalsayed10) [reply workshops/website_mockup_generator/README.md:39]: capitalizations

File: workshops/website_mockup_generator/README.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    <title>repl.it</title>
+    <link href="style.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <script src="script.js"></script>
+  </body>
+</html>
+```
+#### The site title
+In the `<head>` section, Replace the text inside the title tag to the title you want. In my case, I am replacing it with 'Website Mockup Generator'
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:76]:
```suggestion
In the `<head>` section, Replace the text inside the title tag to the title you want. In my case, I am replacing it with 'Website Mockup Generator'.

```

File: workshops/website_mockup_generator/README.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+```html 
+<title>Website Mockup Generator</title>
+```
+
+#### Changing the body background color and alignment
+In order to change the background color and the text alignment inside the body, specify a class called 'body' in the body tag. We will be providing CSS styling for the body class later.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:83]:
```suggestion
In order to change the background color and the text alignment inside the `body`, specify a class called '``body``' in the body tag. We will be providing CSS stylings for the `body` class later.

```

File: workshops/website_mockup_generator/README.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+```html
+<body class="body">
+```
+
+#### The heading
+In the `<body>` provide the heading inside `<h1>...</h1>` tags. In my case, I am providig the heading as 'Website Mockup Generator'.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:90]:
```suggestion
In the `<body>` we will add a heading inside `<h1>...</h1>` tags.

```

Comment (faisalsayed10) [reply workshops/website_mockup_generator/README.md:90]: last line is not required

File: workshops/website_mockup_generator/README.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+
+```html
+<h1>Website Mockup Generator</h1>
+```
+
+#### Placing a text input field
+In order to place a text input field to get the website from the URL, use the following line to create a text input with a placeholder 'Enter the website' and provide id to the input as 'websiteAddress'.
+
+```html
+<input type="text" id="websiteAdress" placeholder='Enter the website'>
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:100]: typo in the id name

File: workshops/website_mockup_generator/README.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+#### Placing the buttons
+We need two buttons to generate the mockup and to download respectively. Use the `<button>` tag twice to place the buttons. Provide the button label in between the `<button></button>` tags. Provide the id as `submitButton` for the 'Generate mockup' button and `Download` for 'Download' button.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:104]:
```suggestion
We need two buttons to generate the mockup and to download respectively. Use the `<button>` tag twice to place the buttons. Provide the button label in between the `<button></button>` tags. Provide the id as `submitButton` for the 'Generate mockup' button and `Download` for the 'Download' button.

```

File: workshops/website_mockup_generator/README.md [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+```html
+<button type="submit" id="submitBtn" onclick='generateMockup();'>Generate Mockup</button>
+<button id="Download">Download</button>
+```
+
+#### Generated mockup image
+We will be using an `<image>` tag to display the generated mockup image. Provide the id as `resultImg`.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:112]: ``

File: workshops/website_mockup_generator/README.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+  </body>
+</html>
+```
+
+And that's it! We have completed writing our HTML.
+
+
+### 2) style.css
+
+The style.css file is used to customize the look of the website mockup generator. We will be making the background color as red using `backgroundcolor: red;`. We will also align the text to center by `text-align: center;`.
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:147]:
```suggestion
The style.css file is used to add styles to our website. We will be making the background color as red using `background-color: red;`. We will also align the text to center by `text-align: center;`.

```

File: workshops/website_mockup_generator/README.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+The script.js file contains the required JavaScript code. The aim of the JavaScript code is: 
+- To get the website address from the text input
+- Generate a mockup for the website by pressing 'Generate Mockup' button
+- Display the mockup as an image
+- Allow users to download the image by pressing 'Download' button
+
+Let us implement the above steps one by one :
+
+#### Getting the elements from HTML in JS by their Id
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:169]: "JavaScript" "id"

File: workshops/website_mockup_generator/README.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+Firstly, we will be getting the elements from HTML into JS by using their Ids using the following code :
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:170]: id not Id

File: workshops/website_mockup_generator/README.md [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+#### Defining the generateMockup() function
+In order to generate the mockups, we will be defining the generateMockup() function which is called when the 'Generate Mockup' button is pressed. Create a function called generateMockup() using:
+
+```javascript
+const generateMockup = () => {
+// Function definition here
+}
+
+Now, assign the value of `websiteAdress` to the value from the text input field using :
```
Comment (faisalsayed10) [inline workshops/website_mockup_generator/README.md:185]: bug here

**Review by faisalsayed10 — CHANGES_REQUESTED** [review]
Hey, just reviewed this workshop. Few things which are of concern:

- The product is not as visually compelling as expected.
- Lack of images/gif throughout the workshop.
- Various capitalization/grammatical/prasing errors.

Sorry to say but we can't accept this workshop in its current state before all the above stated issues are fixed. closing for now, feel free to reopenonce you do all the changes.

File: workshops/website_mockup_generator/README.md (hunk 5 above)
Comment (hackyguru) [reply workshops/website_mockup_generator/README.md:100]: um that was not a typo actually. it was intentional nvm, i will correct the id name

**Pull request closed without merging**